chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -559,14 +626,50 @@ where
         let input0_getter =
             || tx.inputs().get(0).ok_or(ConnectTransactionError::MissingOutputOrSpent);
 
+        let attempt_to_spend_stake = tx
```

Comment:
Instead of these adhoc checks, I'd rather have universal rules that apply to output type, that can here be directly utilized. I feel this isn't clean because it solves a general problem for one specific type. Namely: "A transaction is attempting to spend something it isn't supposed to do, no matter what". A cleaner solution would create a rule that can be tested independently and applied here. 

This applies to this and the next check.

Reply:
Why was this marked resolved?

Reply:
misunderstood your idea, please take a look now